src/psyclone/tests/psyir/nodes/array_mixin_test.py

```diff
@@ -115,6 +117,12 @@ def test_is_upper_lower_bound(fortran_reader):
     assert not array_ref.is_lower_bound(0)
     assert not array_ref.is_upper_bound(0)
 
+    # Return False if the symbol has not associated type information.
+    array_ref = assigns[2].lhs
+    assert type(array_ref.symbol) is not DataSymbol
```

Comment:
use `not isinstance(array_ref.symbol, DataSymbol)` to make pylint happy

Reply:
Oh yes, that's better.